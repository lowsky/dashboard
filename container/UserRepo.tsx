import React, { useContext } from 'react';

import { BranchesWithErrorMessage, UserWithErrorMessage } from '../restinpeace/fetchGithubApi';
import UILibContext from '../components/UILibContext';

interface UserRepoProps {
    github: {
        user?: UserWithErrorMessage;
        repo?: {
            name: string;
            branches: BranchesWithErrorMessage;
        };
    };
    relay?: any;
}

let UserRepo: React.FunctionComponent<UserRepoProps> = (props: UserRepoProps) => {
  if(!props.github) return null;

    const { github } = props;
    // eslint-disable-next-line no-restricted-globals,no-undef
    const { user, repo } = github;
    const { Repo, User, BranchTable } = useContext(UILibContext);

    return (
        <React.Fragment>
            {/*@ts-ignore*/}
          {repo && <Repo repo={repo} />}
            {/*@ts-ignore*/}
          {user && <User user={user} />}
            <div className="column">
                {/*@ts-ignore*/}
              {repo &&<BranchTable repo={repo} />}
            </div>
        </React.Fragment>
    );
};

export default UserRepo;
