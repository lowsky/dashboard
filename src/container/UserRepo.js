import React from 'react';

import Ui from '../components';

const cards = {
    // height: '400px'
};

let UserRepo = props => {
    const { github = {} } = props;
    const { user, repo = { branches: [] } } = github;

    const User = Ui.createUser(props);
    const Repo = Ui.createRepo(props);
    const BranchesTable = Ui.createBranchTable(props);

    return (
        <div className="mdl-grid" style={cards}>
            <div className="mdl-cell mdl-cell--6-col"><Repo repo={repo} /></div>
            <div className="mdl-cell mdl-cell--6-col"><User user={user} /></div>
            <div className="mdl-cell mdl-cell--12-col"><BranchesTable repo={github.repo} /></div>
        </div>
    );
};

export default UserRepo;
