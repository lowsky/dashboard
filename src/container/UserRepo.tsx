import React, { useContext } from 'react';

import { UILibContext } from '../components';
import { BranchesWithErrorMessage, UserWithErrorMessage } from '../restinpeace/fetchGithubApi';

interface UserRepoProps {
    github: {
        user?: UserWithErrorMessage;
        repo?: {
            name: string;
            branches: BranchesWithErrorMessage;
        };
    };
    relay?: any;
}

let UserRepo: React.FunctionComponent<UserRepoProps> = (props: UserRepoProps) => {
    const { github = {} } = props;
    // eslint-disable-next-line no-restricted-globals,no-undef
    const { user, repo } = github;
    const { Repo, User, BranchTable } = useContext(UILibContext);

    return (
        <React.Fragment>
            {/*@ts-ignore*/}
            <Repo repo={repo} />
            {/*@ts-ignore*/}
            <User user={user} />
            <div className="column">
                {/*@ts-ignore*/}
                <BranchTable repo={repo} />
            </div>
        </React.Fragment>
    );
};

export default UserRepo;
