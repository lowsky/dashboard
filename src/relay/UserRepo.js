import Relay from 'react-relay';

import UserRepo from '../container/UserRepo';

import User from './User';
import BranchesTable from './BranchesTable';
import RepoContainer from './Repo';

export default Relay.createContainer(UserRepo, {
    initialVariables: {
        username: 'lowsky',
        ownerUsername: 'lowsky',
        repoName: 'dashboard',
    },
    fragments: {
        github: () => Relay.QL`
            fragment on GithubAPI {
                user(username:$username) {
                    ${User.getFragment('user')}
                }
                repo(ownerUsername: $ownerUsername, name: $repoName) {
                     ${RepoContainer.getFragment('repo')}
                     ${BranchesTable.getFragment('repo')}
                }
            }
        `,
    },
    name: 'UserRepo',
});
