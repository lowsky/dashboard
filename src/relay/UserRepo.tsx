import { createFragmentContainer } from 'react-relay';
import graphql from 'babel-plugin-relay/macro';

import UserRepo from '../container/UserRepo';

export default createFragmentContainer(UserRepo, {
    github: graphql`
        fragment UserRepo_github on GithubAPI
            @argumentDefinitions(
                userName: { type: "String", defaultValue: "lowsky" }
                repoName: { type: "String", defaultValue: "dashboard" }
            ) {
            user(username: $userName) {
                ...User_user
            }
            repo(ownerUsername: $userName, name: $repoName) {
                ...Repo_repo
                ...BranchesTable_repo
            }
        }
    `,
});
