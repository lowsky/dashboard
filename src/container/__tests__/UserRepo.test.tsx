import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import UserRepo from '../UserRepo';

const user = {};
const repo = { branches: [] };
const github = { user, repo };

describe('UserRepo component', () => {
    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<UserRepo />, div);
    });

    it('should render as expected', () => {
        const props = { github };

        const appContainer = renderer.create(<UserRepo {...props} />).toJSON();
        expect(appContainer).toMatchSnapshot();
    });
});
