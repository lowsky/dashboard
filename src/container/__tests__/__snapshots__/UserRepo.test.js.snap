// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepo component should render as expected 1`] = `
Array [
  <div
    className="column"
>
    <h1
        className="title is-3"
    >
        <span
            className="has-text-grey-light"
        >
            Repository
        </span>
    </h1>
    <h3
        className="subTitle is-3"
    >
        <a
            href="https://github.com/undefined/undefined"
        >
            <strong>
                 / 
            </strong>
            <span
                className="icon"
            >
                <i
                    className="fas fa-github"
                />
            </span>
        </a>
    </h3>
</div>,
  <div
    className="card"
>
    <div
        className="card-content"
    >
        <div
            className="media"
        >
            <div
                className="media-left"
            >
                <figure
                    className="image is-48x48"
                >
                    <img
                        alt="avatar"
                        src={undefined}
                    />
                </figure>
            </div>
            <div
                className="media-content"
            >
                <p
                    className="title is-4"
                >
                    ?
                </p>
                <p
                    className="subtitle is-6"
                />
            </div>
        </div>
        <div
            className="content"
        >
            Owner
        </div>
    </div>
</div>,
  <div
    className="column"
>
    <table
        className="table is-bordered is-striped is-hoverable"
    >
        <thead>
            <tr>
                <th
                    className="is-narrow"
                >
                    Branch
                </th>
                <th
                    className=""
                >
                    Commit
                </th>
            </tr>
        </thead>
        <tbody />
    </table>
</div>,
]
`;
